vunit/activehdl_interface.py

```diff
@@ -453,7 +453,7 @@ def __init__(self):
         self.major = None
         self.minor = None
 
-    _version_re = re.compile(r"(?P<major>\d+)\.(?P<minor>\d+)\.\d+\.\d+")
+    _version_re = re.compile(r"(?P<major>\d+)\.(?P<minor>\d+)[a-zA-Z]?\.\d+\.\d+")
```

Comment:
Now that we've had a bug it would be appropriate to add a test case that would have detected this bug. Since the version number is used to compare if one release is newer than another we also need  a way to determine the order between versions like 10.5 and 10.5a. Currently the "a" is just discarded. Are you up to making these additions?

Reply:
Yes I can make these additions. Could you help me to understand why we need to check the letter portion of the version? From the code it seems like we are checking if the version is 10.1 or greater. As a version with a letter is a later release than one without, can we not safely assume that any version with a letter that is greater than 10.1 (e.g. 10.1a, 10.2a, etc) will also be valid? And therefore the letters are just sub-versions like the 3rd and 4th digits, which are also discarded? Thanks.

Reply:
Whether or not to include the letter really depends on what it means. I'm used to the second digit indicating a new functional feature (which VUnit may depend upon). If 10.1a can have a new feature that 10.1 doesn't have we need a way to separate those versions. Not in this particular case but if we were to depend on a feature released in, let say 12.2c, sometime in the future. OTOH, if the letter denotes a bug fix release we can ignore it just like the 3rd and 4th digit. Do you know? If the letter denotes a new feature release I think we should take it into account since it's easy to miss that once it really matters.

Reply:
I do not know if it means feature or bug fix, but its position next to the second digit does make it seem like a feature. I will add some code to capture it in the VersionConsumer as an extra attribute and check it with the tests. I don't know how I would use the new attribute in the supports_vhdl_package_generics class method as it will work with or without the new attribute. Would it be ok to just leave that method as it is, or do you have something in mind?

Reply:
What you can do is to create something like

``` python
class Version(object):
    def __init__(self, major=None, minor=None, minor_letter=None):
        self.major = major
        self.minor = minor
        self.minor_letter = minor_letter
```

and then let `VersionConumer` inherit from that and just add the `__call__` method. To enable comparison between these classes you can use `@total_ordering` on the `Version` class, see https://github.com/VUnit/vunit/blob/master/vunit/vhdl_standard.py for inspiration. With that in place a test case like this should pass:

``` python
def test_comparison(self):
    consumer = VersionConsumer()
    consumer("some version line with 10.5a")
    self.assertTrue(consumer > Version(10, 5))
    self.assertFalse(consumer > Version(10, 5, "a"))
```